packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -289,8 +289,8 @@ class NumberInput extends Component {
             onClick && onClick(evt, { value, direction });
             onChange && onChange(evt, { value, direction });
           } else {
-            onClick && onClick(evt, direction);
-            onChange && onChange(evt, direction);
+            onClick && onClick(evt, { value, direction });
```

Comment:
This might be something that we'll need to wrap behind a feature flag since this will act as a breaking change for the `onClick` prop. In other words, this usage would break with this change:

```js
<NumberInput
  onChange={(event, direction) => {
    if (direction === 'down') {
      // ...
    }
  })}
/>
```